### PR TITLE
Add baseline support for organisation repositories

### LINKED ISSUE #620: Organisations: List organisation repositories on the "add repositories" page, so they can be added to the dashboard [4]
- [ ] Ensure organisation repository information is being pulled from GitHub on sign-up
- [ ] Ensure already added organisations are greyed out
- [ ] List organisation repositories in /select-repositories
- [ ] Ensure organisation repositories can be added to the dashboard
- [ ] We need to ensure that only repositories for which the user is an "admin" are shown

### LINKED ISSUE #648: Organisations: Also show organisation repositories on "my repos" dashboard even if you're not admin [2]

### PR DESCRIPTION
**Pull request by cjwatson**
This shows organisation repositories for which the user has admin
permissions in the add-repositories dialog, and shows organisation
repositories that the user can write to on the my-repositories page.
Various aspects of the UI will need to be tidied up in future PRs, but
this at least gets us started.

We now request the `read:org` scope from GitHub, which allows us to tell
which organisations a user is in.

I also included a change to sort repository list rows alphabetically,
since given the rest of this change I was seeing odd results without
that.

Fixes #620; fixes #648.

### CODE REVIEW

File: src/server/handlers/launchpad.js [hunk 1]
```diff
@@ -411,34 +412,60 @@ export const internalFindSnap = async (repositoryUrl) => {
...
   try {
-    const result = await getMemcached().get(cacheId);
-    if (result !== undefined) {
-      return result.map((entry) => {
+    snaps = await Promise.all(
+      cacheIds.map((cacheId) => memcached.get(cacheId))
+    );
+    remainingPrefixes = zip(urlPrefixes, snaps)
+      .filter(([, snap]) => snap === undefined)
+      .map(([urlPrefix, ]) => urlPrefix);
```
Comment (earnubs) [inline src/server/handlers/launchpad.js]: Consider a new line here, so I can catch my breath :)

(I think the old style guide said new line after assignments)

File: src/server/handlers/launchpad.js [hunk 2]
```diff
@@ -411,34 +412,60 @@ export const internalFindSnap = async (repositoryUrl) => {
...
+    if (!remainingPrefixes.length) {
+      return [].concat(...snaps).map((entry) => {
```
Comment (earnubs) [inline src/server/handlers/launchpad.js]: is this the same as `[...snaps]`?

File: test/routes/src/server/routes/t_github-auth.js [hunk 3]
```diff
@@ -4,9 +4,17 @@ import nock from 'nock';
 import expect from 'expect';
 import url from 'url';
 
-import auth from '../../../../../src/server/routes/github-auth';
 import db from '../../../../../src/server/db';
+import {
+  listOrganizationsCacheId
+} from '../../../../../src/server/handlers/github';
```
Comment (earnubs) [inline test/routes/src/server/routes/t_github-auth.js]: FYI We talked recently in the fortnightly JS discussion about moving tests along side src as a way to resolve this `../../../../../` line noise, something for the future maybe, but maybe we should open that meeting up to members of OLS and further afield.

File: src/server/handlers/launchpad.js (hunk 2 above)
Comment (cjwatson) [inline src/server/handlers/launchpad.js]: No, I'm using this as a one-level flatten operator:

    > var foo = [[1, 2], [3, 4]];
    undefined
    > [].concat(...foo)
    [ 1, 2, 3, 4 ]
    > [...foo]
    [ [ 1, 2 ], [ 3, 4 ] ]

However, you've caused me to realise that this code is confusingly written (each element of `snaps` here is in fact an array of snaps) and possibly wrong in one place, so I'll tidy that up.

File: test/routes/src/server/routes/t_github-auth.js (hunk 3 above)
Comment (cjwatson) [inline test/routes/src/server/routes/t_github-auth.js]: I would be overjoyed to see that change.

**Pull request merged**